static/app/views/onboarding/targetedOnboarding/platform.tsx

```diff
@@ -1,16 +1,41 @@
+import {useEffect, useState} from 'react';
 import styled from '@emotion/styled';
 import {motion} from 'framer-motion';
 
 import ExternalLink from 'sentry/components/links/externalLink';
 import MultiPlatformPicker from 'sentry/components/multiPlatformPicker';
+import {PlatformKey} from 'sentry/data/platformCategories';
 import {t, tct} from 'sentry/locale';
 import testableTransition from 'sentry/utils/testableTransition';
+import useApi from 'sentry/utils/useApi';
 import StepHeading from 'sentry/views/onboarding/components/stepHeading';
 
 import CreateProjectsFooter from './components/createProjectsFooter';
-import {StepProps} from './types';
+import {ClientState, StepProps} from './types';
 
 function OnboardingPlatform(props: StepProps) {
+  const [selectedPlatforms, setSelectedPlatforms] = useState<PlatformKey[]>([]);
+  const addPlatform = (platform: PlatformKey) => {
+    setSelectedPlatforms([...selectedPlatforms, platform]);
+  };
+  const removePlatform = (platform: PlatformKey) => {
+    setSelectedPlatforms(selectedPlatforms.filter(p => p !== platform));
+  };
+
+  const api = useApi();
+  useEffect(() => {
+    api
```

Comment:
@Neo-Zhixing we're gonna have a lot of hooks that are calling the `/client-state/onboarding/` endpoint when we add in integrations. I wonder if we can improve code-sharing a bit more here. As well as automatically adding the default value for `platform` if we get no response from the server. Too much code like `if (state.platforms)`

Reply:
Created `fetchClientState`